mora/service.py

```diff
@@ -0,0 +1,415 @@
+#
+# Copyright (c) 2017, Magenta ApS
+#
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+#
+
+import enum
+import operator
+
+import flask
+import iso8601
+import werkzeug
+
+from . import lora, util
+
+blueprint = flask.Blueprint('service', __name__, static_url_path='',
+                            url_prefix='/service')
+
+
+def _get_connector():
+    args = flask.request.args
+
+    loraparams = dict()
+
+    if args.get('at'):
+        loraparams['effective_date'] = iso8601.parse_date(args['at'])
+
+    return lora.Connector(**loraparams)
+
+
+@blueprint.route('/o/')
+@util.restrictargs('at')
+def list_organisations():
+    '''List displayable organisations. We consider anything that has *at
+    least one* root unit 'displayable'.
+
+    .. :quickref: Organisation; List
+
+    :queryparam date at: Current time in ISO-8601 format.
+
+    :<jsonarr string name: Human-readable name of the organisation.
+    :<jsonarr string uuid: Machine-friendly UUID of the organisation.
+
+    :status 200: Always.
+
+    **Example Response**:
+
+    .. sourcecode:: json
+
+      [
+        {
+          "name": "Aarhus Kommune",
+          "uuid": "59141156-ed0b-457c-9535-884447c5220b"
+        },
+        {
+          "name": "Ballerup Kommune",
+          "uuid": "3a87187c-f25a-40a1-8d42-312b2e2b43bd"
+        }
+      ]
+
+    '''
+    c = _get_connector()
+
+    r = []
+
+    for orgid, org in c.organisation.get_all(bvn='%'):
+        attrs = org['attributter']['organisationegenskaber'][0]
+
+        children = c.organisationenhed(overordnet=orgid, gyldighed='Aktiv')
+
+        if len(children):
+            r.append({
+                'name': attrs['organisationsnavn'],
+                'user_key': attrs['brugervendtnoegle'],
```

Comment:
Bruger vi overhovedet user_key? (Den fremgår heller ikke af 'Example response')

Reply:
Nej, men den er jo gratis at have med, og derudover så kunne den være brugbar for andre brugere af APIet — hvis fx Ballerup ønskede at bruge det.